test/test_profile_mp_mnist.py

```diff
@@ -129,8 +129,9 @@ def train_mnist(flags,
   optimizer = optim.SGD(model.parameters(), lr=lr, momentum=flags.momentum)
   loss_fn = nn.NLLLoss()
 
-  # Start up client side profiler server.
-  server = xp.start_server(flags.profiler_port)
+  # Only start up profiler server once *per* host.
+  if xm.is_master_ordinal():
+    server = xp.start_server(flags.profiler_port)
```

Comment:
can we roll the if check into the start_server function? seems easier to use that way.

Reply:
If user wants to profile multiple client side processes, the should be able to start servers on different processes not only master. The fix should be on TF server implementation side by xprof team, which they're working on supporting long-term.

Reply:
maybe the func can accept ordinals to run on then? you can default to ordinal0 only.